### PR TITLE
refactor(nlp): Config Reorganization

### PR DESCRIPTION
**Pull request by elozano98**
## Description
Configuration of main NLP task classes have been refactored into a better structure. Also, duplicated code has been removed.  

## Approach taken / Explain the design
- `NerConfig` and `IntentClassifier` config are extended from `NlpTaskConfig` and don't include the preprocess engine.
- `IntentClassificationConfigStorage` and `NerConfigStorage` have been replaced by a general `ConfigStorage` class.
- `BotonicIntentClassifier` and `BotonicNer` constructor have now two params: `configuration` and `preprocessor`.
- The configuration types created include the following fields:
```typescript
export interface NlpTaskConfig {
  locale: Locale
  maxLength: number
  vocabulary: string[]
}
```

```typescript
export interface IntentClassifierConfig extends NlpTaskConfig {
  intents: string[]
}
```

```typescript
export interface NerConfig extends NlpTaskConfig {
  entities: string[]
}
```

## Testing

The pull request...

- [x] has unit tests
- [x] has integration tests

### CODE REVIEW

File: packages/botonic-nlp/src/storage/config-storage.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export interface NlpTaskConfig {
+  locale: Locale
+  maxLength: number
+  vocabulary: string[]
+}
+
+export class ConfigStorage {
+  public readonly CONFIG_FILENAME = 'config.json'
+
+  load<T extends NlpTaskConfig>(path: string): T {
```
Comment (dpinol) [inline packages/botonic-nlp/src/storage/config-storage.ts:16]: nice generics! :clap:

**Review by ericmarcos — APPROVED** [review]

**Review by dpinol — CHANGES_REQUESTED** [review]
:clap: try the Config Storage generic change and go!

File: packages/botonic-nlp/src/storage/config-storage.ts (hunk 1 above)
Comment (elozano98) [reply packages/botonic-nlp/src/storage/config-storage.ts]: Done! 🚀

Comment (dpinol) [reply packages/botonic-nlp/src/storage/config-storage.ts]: try:
`type NerConfigStorage = ConfigStorage<NerConfig>`

**Review by dpinol — APPROVED** [review]

**Pull request merged**